### PR TITLE
Suggestion for adding model option

### PR DESCRIPTION
**Pull request by azhx**
Just a suggestion to add an option for specifying the model as an environment variable, rather than hardcoding text-davinci-003.

### CODE REVIEW

**Comment by azhx** [comment]
maybe in the future add something to error if the model spec doesn't exist, idk, i spent like 30 min on this

File: src/openai_cli/cli.py [hunk 1]
```diff
@@ -14,23 +14,31 @@ def cli():
...
+    client = build_completion_client(token = get_token(token), model = get_model(model))
     prompt = source.read()
     result = client.generate_response(prompt)
     click.echo(result)
 
 
 @cli.command()
 @click.option("-t", "--token", default="", help="OpenAI API token")
-def repl(token: str) -> None:
+@click.option('-m', "--model", default="", help="OpenAI model option")
```
Comment (peterdemin) [inline src/openai_cli/cli.py:28]: text-davinci-003 is the most advanced model, I think it's to have it as a default here.

File: src/openai_cli/cli.py [hunk 2]
```diff
@@ -14,23 +14,31 @@ def cli():
...
     """Start interactive shell session for OpenAI completion API."""
-    client = build_completion_client(get_token(token))
+    client = build_completion_client(token = get_token(token), model = get_model(model))
     while True:
         print(client.generate_response(input("Prompt: ")))
         print()
 
+def get_model(model: str) -> str:
+    if not model:
+        model = os.environ.get("OPENAI_MODEL", "")
```
Comment (peterdemin) [inline src/openai_cli/cli.py:38]: Let's drop the environment option here. I find it confusing to have this configured through env.

File: src/openai_cli/cli.py [hunk 3]
```diff
@@ -14,23 +14,31 @@ def cli():
...
+    if not model: 
+        print("OPENAI_MODEL environment variable not set. Defaulting to text-davinci-003")
```
Comment (peterdemin) [inline src/openai_cli/cli.py:40]: Without the env var support, this print statement can be dropped too.

File: src/openai_cli/client.py [hunk 4]
```diff
@@ -7,11 +7,14 @@ class CompletionClient:
     TEMPERATURE = 0.1
     TIMEOUT = 60
 
-    def __init__(self, token: str, session: requests.Session) -> None:
+    def __init__(self, token: str, model:str, session: requests.Session) -> None:
         self._headers = {"Authorization": f"Bearer {token}"}
         self._session = session
+        self.model = model
 
     def generate_response(self, prompt: str, model: str = "text-davinci-003") -> str:
```
Comment (peterdemin) [inline src/openai_cli/client.py:15]: Let's drop the model parameter here too, we don't need to have two ways to pass it.

**Review by peterdemin — CHANGES_REQUESTED** [review]

**Review by peterdemin — APPROVED** [review]

**Comment by peterdemin** [comment]
Thanks!

**Comment by peterdemin** [comment]
Released 0.0.2: https://github.com/peterdemin/openai-cli/blob/main/CHANGES.rst#002---dec-29-2022

**Pull request merged**